Objective-C/CBLFragment.h

```diff
@@ -0,0 +1,44 @@
+//
+//  CBLFragment.h
+//  CouchbaseLite
+//
+//  Created by Pasin Suriyentrakorn on 4/21/17.
+//  Copyright © 2017 Couchbase. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "CBLArray.h"
+#import "CBLArrayFragment.h"
+#import "CBLDictionaryFragment.h"
+#import "CBLReadOnlyFragment.h"
+#import "CBLSubdocument.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/** CBLFragment protocol provides read and write access to the data value wrapped by 
+    a fragment object. */
+@protocol CBLFragment <CBLReadOnlyFragment>
+
+/** Gets or sets the value as an object. The object types are CBLBlob, CBLReadOnlyArray, 
+    CBLReadOnlyDictionary, NSNumber, or NSString based on the underlying data type; or nil 
+    if the value is nil. */
+@property (nonatomic, nullable) NSObject* value;
```

Comment:
We have those in CBLReadOnlyFragment. The value property is mainly for setting as setting to object is weird.